ee/server/scim/groups.go

```diff
@@ -0,0 +1,327 @@
+		return scim.Resource{}, err
+	}
+
+	return createGroupResource(group), nil
+}
+
+func createGroupFromAttributes(attributes scim.ResourceAttributes) (*fleet.ScimGroup, error) {
+	group := fleet.ScimGroup{}
+	var err error
+	group.DisplayName, err = getRequiredResource[string](attributes, displayNameAttr)
+	if err != nil {
+		return nil, err
+	}
+	group.ExternalID, err = getOptionalResource[string](attributes, externalIdAttr)
+	if err != nil {
+		return nil, err
+	}
+
+	// Process members
+	members, err := getComplexResourceSlice(attributes, membersAttr)
+	if err != nil {
+		return nil, err
+	}
+	userIDs := make([]uint, 0, len(members))
+	for _, member := range members {
+		// Get the value attribute which contains the user ID
+		valueIntf, ok := member["value"]
+		if !ok || valueIntf == nil {
+			continue
+		}
+		valueStr, ok := valueIntf.(string)
+		if !ok {
+			return nil, errors.ScimErrorBadParams([]string{"value"})
+		}
+
+		// Extract user ID from the value
+		userID, err := extractUserIDFromValue(valueStr)
+		if err != nil {
+			return nil, errors.ScimErrorBadParams([]string{"value"})
+		}
+		userIDs = append(userIDs, userID)
+	}
+	group.ScimUsers = userIDs
+
+	return &group, nil
+}
+
+// Get the Scim group by ID. The group id is of the format: group-123
+// SCIM resource IDs must be unique across all resources.
+func (g *GroupHandler) Get(r *http.Request, id string) (scim.Resource, error) {
+	idUint, err := extractGroupIDFromValue(id)
+	if err != nil {
+		level.Info(g.logger).Log("msg", "failed to parse id", "id", id, "err", err)
+		return scim.Resource{}, errors.ScimErrorResourceNotFound(id)
+	}
+
+	group, err := g.ds.ScimGroupByID(r.Context(), idUint)
+	switch {
+	case fleet.IsNotFound(err):
+		level.Info(g.logger).Log("msg", "failed to find group", "id", id)
+		return scim.Resource{}, errors.ScimErrorResourceNotFound(id)
+	case err != nil:
+		level.Error(g.logger).Log("msg", "failed to get group", "id", id, "err", err)
+		return scim.Resource{}, err
+	}
+
+	return createGroupResource(group), nil
+}
+
+func createGroupResource(group *fleet.ScimGroup) scim.Resource {
+	groupResource := scim.Resource{}
+	groupResource.ID = scimGroupID(group.ID)
+	if group.ExternalID != nil {
+		groupResource.ExternalID = optional.NewString(*group.ExternalID)
+	}
+	groupResource.Attributes = scim.ResourceAttributes{}
+	groupResource.Attributes[displayNameAttr] = group.DisplayName
+
+	// Add members if any
+	if len(group.ScimUsers) > 0 {
+		members := make([]scim.ResourceAttributes, 0, len(group.ScimUsers))
+		for _, userID := range group.ScimUsers {
+			members = append(members, map[string]interface{}{
+				"value": scimUserID(userID),
+				"$ref":  "Users/" + scimUserID(userID),
+				"type":  "User",
+			})
+		}
+		groupResource.Attributes[membersAttr] = members
+	}
+
+	return groupResource
+}
+
+func (g *GroupHandler) GetAll(r *http.Request, params scim.ListRequestParams) (scim.Page, error) {
+	page := params.StartIndex
+	if page < 1 {
+		page = 1
```

Comment:
I have a vague sense that `ListOptions` in other parts of the codebase use a zero-indexed approach, but I don't know if it has been consistently applied as a pattern. Perhaps it would be worth documenting this somewhere?